src/Libraries/Microsoft.Extensions.TimeProvider.Testing/FakeTimeProvider.cs

```diff
@@ -7,88 +7,125 @@
 using System.Diagnostics.CodeAnalysis;
 using System.Globalization;
 using System.Threading;
+using Microsoft.Extensions.Time.Testing;
 using Microsoft.Shared.Diagnostics;
 
 namespace Microsoft.Extensions.Time.Testing;
 
 /// <summary>
-/// A synthetic clock used to provide deterministic behavior in tests.
+/// A synthetic time provider used to enable deterministic behavior in tests.
 /// </summary>
 public class FakeTimeProvider : TimeProvider
 {
-    internal readonly List<FakeTimeProviderTimer.Waiter> Waiters = new();
+    internal readonly HashSet<Waiter> Waiters = new();
```

Comment:
Should this not be private and named `waiters`?

Reply:
It's used by tests, so it needs to be internal.

Reply:
Ahh right. I do remember that. Not a big fan of depending on internal state in a test though. Would suggest changing that but that's for another PR.